### DISCUSSION TITLE
Ergonomic way to combine parsers of different types?

### DISCUSSION THREAD

**Discussion by npetrangelo** [winnow-rs/winnow#621] [Q&A] [answered]
I want to do something like this, but the Map type seems to have too much type information to map everything to the same type for `alt()`. How should I do this?
```rust
pub fn parse_value(s: &mut &str) -> PResult<Value> {
    alt([
        parse_string.map(|s| Value::String(s)),
        parse_float.map(|f| Value::Float(f)),
        parse_int.map(|i| Value::Int(i)),
        parse_bool.map(|b| Value::Bool(b)),
        parse_range.map(|r| Value::Range(r))])
    .parse_next(s)
}
```

**Comment by epage** [comment, marked as answer]
Instead of `[]` in the `alt`, use `()` which allows different types. Thats the primary way we recommend people use `alt`. ‘[]` support is newer and more for some corner cases.
